### DISCUSSION TITLE
Docker for testing

### DISCUSSION THREAD

**Discussion by aarcex3** [piccolo-orm/piccolo#1100] [Q&A] [answered]
Hi, I'm trying to implement feature #1075.
The Piccolo documentation mentions having a local database for CockroachDB and PostgreSQL. Is it possible to use Docker instead? If so, how can I set it up?
I'm currently having issues getting it to work.

**Comment by dantownsend** [comment, marked as answer]
If using Docker, you could:

* Run it using host networking
* Or expose port 5432 to your system

I usually just run Postgres on the bare metal for local testing, using Postgres.app on the Mac, or just `apt install` on Ubuntu. I can't really help with Windows.

**Reply by pavdwest** [reply]
@aarcex3 I have a simple dockerised starter example here for reference: https://github.com/pavdwest/docker_litestar_piccolo_pg_starter
